lib/active_model/serializer/builder.rb

```diff
@@ -0,0 +1,71 @@
+require "set"
+module ActiveModel
+  class Serializer
+    class Builder
+      extend ActiveSupport::Autoload
+
+      ADAPTER_OPTION_KEYS = Set.new([:include, :fields, :adapter])
+
+      def initialize(resource, options = {})
+        @resource = resource
+        @adapter_opts, @serializer_opts =
+          options.partition { |k, _| ADAPTER_OPTION_KEYS.include? k }.map { |h| Hash[h] }
+      end
+
+      delegate :serializable_hash, :as_json, :to_json, to: :adapter
```

Comment:
convenience methods. Calling ActiveModel::Serializer.build without a block returns the builder, but you can still call to_json, as_json, serializable_hash on it.